### PR TITLE
Provide symbol outline for psd1 files

### PR DESCRIPTION
**Pull request by kapilmb**
This update will show the list of keys present in a psd1 file when a user invokes `Go to Symbols in File... (Ctrl+Shift+O)` on an psd1 file.

### CODE REVIEW

File: src/PowerShellEditorServices/Language/FindSymbolsVisitor.cs [hunk 1]
```diff
@@ -89,4 +89,48 @@ private bool IsAssignedAtScriptScope(VariableExpressionAst variableExpressionAst
             return false;
         }
     }
+
+    internal class FindHashtabeSymbolsVisitor : AstVisitor
```
Comment (daviwil) [inline src/PowerShellEditorServices/Language/FindSymbolsVisitor.cs]: Slight typo, **FindHashtabeSymbolsVisitor**

**Comment by kapilmb** [comment]
@daviwil rebased on develop, fixed the type and updated inline docs.

**Comment by daviwil** [comment]
This is pretty awesome!  Really handy to jump down to a specific field in the file, even in nested structures.  It'd be really nice if we could also jump to commented-out fields too but based on some brief research there seems to be no Ast class or visitor for comment lines.  Any thoughts on how we might be able to do that?

**Comment by daviwil** [comment]
Hmmm, we might be able to use the list of tokens returned by the parser to find them.  There's a `TokenKind.Comment` here: https://msdn.microsoft.com/en-us/library/system.management.automation.language.tokenkind%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396

**Comment by rkeithhill** [comment]
> there seems to be no Ast class or visitor for comment lines

I've asked Jason about this. This is one area where I wish PowerShell were more like Roslyn which retains all source code trivia (like whitespace between tokens and comments).

**Comment by daviwil** [comment]
I've considered creating a higher-level abstraction that can (literally) fill in the gaps of the Ast for things like that.  I think we're going to need it for making code formatting rules in PSScriptAnalyzer.

**Comment by kapilmb** [comment]
I think we can tackle some basic scenarios without needing any help from Asts. For example, 
```powershell
@{
  ModuleVersion = 0.1
  Author = "XYZ"
 # RootModule = ".\test.psm1"
 # FunctionsToExport = @('get-foo')
}
```
Here, we just extract out the commented fields with some regular expression help.

**Comment by daviwil** [comment]
@kapilmb I think this PR is done, right?  Can I merge it?

**Comment by kapilmb** [comment]
Yes, this done.

**Comment by daviwil** [comment]
Cool!  Merging it.

**Pull request merged**